lib/protest/runner.rb

```diff
@@ -42,6 +42,10 @@ def report(test)
     def fire_event(event, *args)
       event_handler_method = :"on_#{event}"
       @report.send(event_handler_method, *args) if @report.respond_to?(event_handler_method)
+
+      if Protest.fail_fast? && [:failure, :error].include?(event)
```

Comment:
I'd prefer to exit when the exception is detected and rescued (https://github.com/matflores/protest/blob/20960cbf46a5efdc5802f8c785176c5f84a06894/lib/protest/runner.rb#L33-L37) instead of doing it in this method which gets called after any kind of event.

@fsaravia thoughts?